bokehjs/src/coffee/source/geojson_data_source.coffee

```diff
@@ -0,0 +1,127 @@
+
+class GeoJSONDataSource extends ColumnDataSource.Model
+
+  initialize: (options) ->
+    super(options)
+    @register_property('data', @geojson_to_column_data, true)
+    @add_dependencies('data', this, ['geojson'])
+    
+  nonserializable_attribute_names: () ->
+    super().concat(['data'])
+
+  _get_new_list_array: (length) ->
+    array = new Array(length)
+    list_array = _.map(array, (x) -> [])
+    return list_array
+
+  _get_new_nan_array: (length) ->
+    array = new Array(length)
+    nan_array = _.map(array, (x) -> NaN)
+    return nan_array
+
+  geojson_to_column_data: () ->
+    geojson = JSON.parse(@get('geojson'))
+
+    if geojson.type not in ['GeometryCollection', 'FeatureCollection']
+      throw new Error('Bokeh only supports type GeometryCollection and FeatureCollection at top level')
+
+    if geojson.type == 'GeometryCollection' 
+      if not geojson.geometries?
+        throw new Error('No geometries found in GeometryCollection')
+      if geojson.geometries.length == 0
+        throw new Error('geojson.geometries must have one or more items')
+      items = geojson.geometries
+
+    if geojson.type == 'FeatureCollection'
+      if not geojson.features?
+        throw new Error('No features found in FeaturesCollection')
+      if geojson.features.length == 0
+        throw new Error('geojson.features must have one or more items')
+      items = geojson.features
+    
+    data = {
+      'x': @_get_new_nan_array(items.length),
+      'y': @_get_new_nan_array(items.length),
+      'z': @_get_new_nan_array(items.length),
+      'xs': @_get_new_list_array(items.length),
+      'ys': @_get_new_list_array(items.length),
+      'zs': @_get_new_list_array(items.length)
+    }
+
+    # The flatten function is used to link together MultiLines and MultPolygons with NaNs
+    flatten_function = (accumulator, currentItem) ->
+      return accumulator.concat([[NaN, NaN, NaN]]).concat(currentItem)
+
+    for item, i in items
+
+
+      if item.type == 'Feature'
+        geometry = item.geometry
+
+        # Only Features have properties
+        for property of item.properties
+          if !data.hasOwnProperty(property)
+            data[property] = @_get_new_nan_array(items.length)
+          data[property][i] = item.properties[property]
+
+      else
+        geometry = item
+
+      # Now populate based on Geometry type
+
+      switch geometry.type
+
+        when "Point"
+          coords = geometry.coordinates
+          data.x[i] = coords[0]
+          data.y[i] = coords[1]
+          data.z[i] = coords[2] ? NaN
+
+        when "LineString"
+          coord_list = geometry.coordinates
+          for coords, j in coord_list
+            data.xs[i][j] = coords[0]
+            data.ys[i][j] = coords[1]
+            data.zs[i][j] = coords[2] ? NaN
+
+        when "Polygon"
+          if geometry.coordinates.length > 1
+            logger.warn('Bokeh does not support Polygons with holes in, only exterior ring used.')
```

Comment:
This reads very oddly to me, "in the interior" would be better I think. Doesn't the other PR add support for holes? Would it make sense to wait on that one and then update this one?